charms/worker/k8s/src/token_distributor.py

```
@@ -299,7 +299,7 @@ def remove(self, name: str, secret: Optional[ops.Secret], ignore_errors: bool):
             K8sdConnectionError: reraises cluster token remove failures
         """
         try:
-            self.api_manager.remove_node(name)
+            self.api_manager.remove_node(name, force=ignore_errors)
```

we only need to force if we intentionally are ignoring errors